### ISSUE TITLE
Revenue

### ISSUE THREAD

**Issue by daniestepona** [daniestepona/Pharma-Revenue-Dashboard-#1]
import streamlit as st
import pandas as pd
import numpy as np
import plotly.express as px
from sklearn.linear_model import LinearRegression
from statsmodels.tsa.arima.model import ARIMA

# Streamlit App Title
st.title("📊 AI-Driven Pharma Revenue Forecasting Dashboard")

# File Upload Section
st.sidebar.header("Upload Revenue Data")
uploaded_file = st.sidebar.file_uploader("Upload Excel File", type=["xlsx"])

if uploaded_file:
    # Read Excel File
    df = pd.read_excel(uploaded_file)

    # Ensure Correct Column Names
    required_columns = ["Month", "Revenue"]
    if not all(col in df.columns for col in required_columns):
        st.error("Invalid format. Ensure columns: 'Month' and 'Revenue'.")
    else:
        # Convert Month to String (Ensure Proper Sorting)
        df["Month"] = df["Month"].astype(str)
        
        # Display Uploaded Data
        st.subheader("📄 Uploaded Revenue Data")
        st.dataframe(df)

        # Trend Visualization
        st.subheader("📈 Revenue Trends")
        fig = px.line(df, x="Month", y="Revenue", markers=True, title="Revenue Over Time")
        st.plotly_chart(fig)

        # Prepare Data for Forecasting
        df["Month_Num"] = np.arange(len(df))
        X = df[["Month_Num"]]
        y = df["Revenue"]

        # AI-Based Forecasting
        model = LinearRegression()
        model.fit(X, y)
        future_months = np.arange(len(df), len(df) + 6).reshape(-1, 1)
        predicted_revenue = model.predict(future_months)

        # ARIMA Model
        arima_model = ARIMA(df["Revenue"], order=(2, 1, 2))
        arima_model_fit = arima_model.fit()
        arima_predictions = arima_model_fit.forecast(steps=6)

        # Create Forecast DataFrame
        forecast_months = [f"Month {i}" for i in range(len(df) + 1, len(df) + 7)]
        forecast_df = pd.DataFrame({
            "Month": forecast_months,
            "Linear Regression Prediction": predicted_revenue.astype(int),
            "ARIMA Prediction": arima_predictions.astype(int)
        })

        # Display Forecasts
        st.subheader("📊 AI Revenue Forecasts")
        st.dataframe(forecast_df)

        # Visualization of Predictions
        st.subheader("🔍 AI Forecasting Comparisons")
        fig_forecast = px.line(forecast_df, x="Month", y=["Linear Regression Prediction", "ARIMA Prediction"],
                               markers=True, title="AI-Based Revenue Predictions")
        st.plotly_chart(fig_forecast)

        # Download Option
        st.sidebar.subheader("📥 Download Reports")
        forecast_excel = forecast_df.to_excel("Pharma_Revenue_Forecast.xlsx", index=False)
        st.sidebar.download_button(label="📊 Download Forecast Report (Excel)", 
                                   data=open("Pharma_Revenue_Forecast.xlsx", "rb"), 
                                   file_name="Pharma_Revenue_Forecast.xlsx", 
                                   mime="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet")
